### PR TITLE
Visualize dependencies/imports

### LINKED ISSUE #43: Show dependencies through lines (only appear on hover)

### PR DESCRIPTION
**Pull request by Dara2004**
close #43 #40 #86 
There are not many dependencies so should we just show all the arrows instead of only show them on hover?

### CODE REVIEW

**Comment by MayKikuchi** [comment]
screen
 there seems to be an issue with the dependencies getting cut off at the bottom

also, the extra scroll bar cutting off the screen on the right side looks a bit odd - I have a PR up to handle scrolling, maybe that could be helpful?

**Comment by Dara2004** [comment]
> * there seems to be an issue with the dependencies getting cut off at the bottom
> * the file names are on top of the islands, so they cover some of the icons (e.g. the STATEMENT flag), making the icon un-hoverable
> * also, the extra scroll bar cutting off the screen on the right side looks a bit odd - I have a PR up to handle scrolling, maybe that could be helpful?

@MayKikuchi I fixed the first two issues. Not sure how to fix the right scroll bar since I can't replicate the issue anymore...
image

**Comment by MayKikuchi** [comment]
> > * there seems to be an issue with the dependencies getting cut off at the bottom
> > * the file names are on top of the islands, so they cover some of the icons (e.g. the STATEMENT flag), making the icon un-hoverable
> > * also, the extra scroll bar cutting off the screen on the right side looks a bit odd - I have a PR up to handle scrolling, maybe that could be helpful?
> 
> @MayKikuchi I fixed the first two issues. Not sure how to fix the right scroll bar since I can't replicate this issue...
> image

the right scroll bar thing seems to have gone away! thanks Dara :D

File: client/src/components/Island/index.tsx [hunk 1]
```diff
@@ -30,13 +32,16 @@ const IslandImage = styled.img<{ maxWidth }>`
...
+  left: ${(props) => props.x}px;
+  top: ${(props) => props.y}px;
   display: flex;
   flex-direction: column;
   justify-content: space-between;
   align-items: center;
-  left: ${(props) => `${props.x + legendWidth}px`};
+  left: ${(props) => `${props.x}px`};
   top: ${(props) => `${props.y}px`};
   position: absolute;
```
Comment (MayKikuchi) [inline client/src/components/Island/index.tsx:46]: there are 2 position: absolute in this component

File: client/src/components/Legend/index.tsx [hunk 2]
```diff
@@ -14,7 +14,7 @@ import StoneFence from "../../assets/fences/stoneFence.png";
 import WoodenFence from "../../assets/fences/woodenFence.png";
 import styles from "./Legend.module.css";
 import ColoredFlag, { FlagType } from "../ColoredFlag";
-
+import Airplane from "../../assets/icons/airplane.png";
```
Comment (MayKikuchi) [inline client/src/components/Legend/index.tsx:17]: I love how this Airplane is a helicopter 😂 You totally don't need to change it or anything, I just thought it was funny 🤣

File: client/src/components/RightSide/index.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+const RightSide = ({ javaProject }: { javaProject?: JavaArchipelago }) => {
+  const islands = javaProject.islands;
+  const links = javaProject.links;
+
+  return (
+    <div
+      className={styles.rightSide}
+      style={{
+        height: `${javaProject.height + 150}px`,
```
Comment (MayKikuchi) [inline client/src/components/RightSide/index.tsx:19]: where does the 150 come from? Might be nice to use a constant, not necessary though I guess :P (same with the 150's below)

Comment (Dara2004) [reply client/src/components/RightSide/index.tsx:19]: Yeah that's not very nice... I just wanted to add some px to the height and width to make the right side looks nicer, then ended up changing in a few places. It's not desirable but maybe just leave them there for now :P

**Pull request merged**